File: tree/BasicStructure.cpp
#include<bits/stdc++.h>
using namespace std;
class Node{
    public:
    int data;
    Node *left;
    Node *right;
    Node( int x ){
        data = x;
        left = NULL;
        right = NULL;
    }
};
int main(){
        //                  52
        //          25               96
        //      41      20      78      23
        //    36  48   49 67   76 84   19 12

    Node *root =  new Node(52);
    root->left = new Node(25);
    root->left->left = new Node(41);
    root->left->right = new Node(20);

    root->left->left->left = new Node(36);
    root->left->left->right = new Node(48);

    root->left->right->left = new Node(49);
    root->left->right->right = new Node(67);
    root->right = new Node(96);
    root->right->right = new Node(23);
    root->right->left = new Node(78);

    root->right->right->right = new Node(12);
    root->right->right->left = new Node(19);

    root->right->left->left = new Node(76);
    root->right->left->right = new Node(84);



}

File: tree/binary search tree/java implementation/basicStructure.java
public class basicStructure{
    
    static class Node{
        int data;
        Node left;
        Node right;
        public Node(int data){
            this.data = data;
            left = null;
            right = null;
        }
    }
    public void insert(Node node, int val){
        if(val<node.data){
            if(node.left != null){
                insert(node.left,val);
            }
            else{
                System.out.println("Inserted " + val + " to the left of " + node.data);
                node.left = new Node(val);
            }
        }
        else if(val>node.data){
            if(node.right != null){
                insert(node.right,val);
            }
            else{
                System.out.println("Inserted " + val + " to the right of " + node.data);
                node.right = new Node(val);
            }

        }
    }
    public void inorder(Node node){
        if(node != null){
            inorder(node.left);
            System.out.println(node.data + " ");
            inorder(node.right);
        }   
    }
    public static void main(String args[]){
        basicStructure tree = new basicStructure();
        Node root  = new Node(5);
        tree.insert(root, 2);
        tree.insert(root, 4);
        tree.insert(root, 5);
        tree.insert(root, 6);
        tree.insert(root, 7);
        tree.insert(root, 1);
        tree.insert(root, 3);
        tree.insert(root, 9);
        tree.inorder(root);
    }
}